api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -287,4 +288,28 @@ public void savePatientQueue_shouldNotCompletePatientQueueOnEdit() throws Except
         Assert.assertEquals("Non-Emergency", editedPatientQueue.getPriorityComment());
         Assert.assertEquals(PatientQueue.Status.PENDING, editedPatientQueue.getStatus());
     }
+
+	@Test
+	public void generateVisitNumber_shouldNotThrowOutOfIndexExceptionWhenPreviousQueueLengthLessThanStandardLength() {
+		PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+
+		Patient patient = Context.getPatientService().getPatient(10000);
+
+		Location location = Context.getLocationService().getLocation(1);
+
+		PatientQueue patientQueue = new PatientQueue();
+		patientQueue.setPatient(patient);
+		patientQueue.setStatus(PatientQueue.Status.PENDING);
+		patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
+		patientQueue.setLocationFrom(location);
+		patientQueue.setLocationTo(location);
+		patientQueue.setVisitNumber("20/10/2019-002");
+		patientQueueingService.savePatientQue(patientQueue);
+
+		String visitNumber = patientQueueingService.generateVisitNumber(location, patient);
+
+		Assert.assertNotEquals(STANDARD_VISIT_NUMBER_LENGTH,patientQueue.getVisitNumber());
+
+		Assert.assertEquals(STANDARD_VISIT_NUMBER_LENGTH, visitNumber.length());
```

Comment:
What has this assertion got to do with this test?

Reply:
Its confirming that the previous  visitNumber is actually less that the standard.

Reply:
I mean this: Assert.assertEquals(STANDARD_VISIT_NUMBER_LENGTH, visitNumber.length());